pandas/indexes/multi.py

```diff
@@ -1173,9 +1173,57 @@ def from_product(cls, iterables, sortorder=None, names=None):
 
         labels, levels = _factorize_from_iterables(iterables)
         labels = cartesian_product(labels)
+        return MultiIndex(levels, labels, sortorder=sortorder, names=names)
 
-        return MultiIndex(levels=levels, labels=labels, sortorder=sortorder,
-                          names=names)
+    def _reconstruct(self, sort=False):
+        """
+        reconstruct the MultiIndex
```

Comment:
Can you specify what reconstruct means ? (rearranging the labels and levels to have the levels sorted?)